dataset/models/tf/tf_model.py

```diff
@@ -270,19 +276,26 @@ def _make_inputs(self):
 
             dtype = input_config.get('dtype')
             tensor = tf.placeholder(dtype, name=input_name)
+            output_before[input_name] = tensor
             tensor = self._make_transform(tensor, input_config)
 
             if isinstance(shape, (list, tuple)):
                 tensor = tf.reshape(tensor, [-1] + list(shape))
 
             name = input_config.get('name')
+
+            if name == input_name:
```

Comment:
check for all names and input_names as a output tensor can have the same name as some other placeholder.